community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -458,10 +460,20 @@ private void commit() throws TransactionFailureException
 
         try ( CommitEvent commitEvent = transactionEvent.beginCommitEvent() )
         {
-            // Trigger transaction "before" hooks
-            if ( (hooksState = hooks.beforeCommit( txState, this, storeLayer )) != null && hooksState.failed() )
+            // Trigger transaction "before" hooks.
+            if ( hasChanges() )
```

Comment:
Will that mean that in case of read only transaction your afterCommit or afterRollback hook will never be invoked?